src/NuGet.Core/NuGet.Common/Migrations/Migration1.cs

```diff
@@ -79,7 +79,8 @@ void AddAllParentDirectoriesUpToHome(string path)
             {
                 pathsToCheck.Add(path);
 
-                if (!path.StartsWith(homePath, StringComparison.Ordinal))
+                // Unix-based systems support only the forward slash which is returned by Path.AltDirectorySeparatorChar
+                if (!path.StartsWith(homePath + Path.AltDirectorySeparatorChar, StringComparison.Ordinal))
```

Comment:
This change hopefully addresses https://github.com/NuGet/NuGet.Client/commit/0966444b35d8a74fa25adf0368b983f389884377#r86522996 comment

Reply:
> Unix-based systems support only the forward slash which is returned by Path.AltDirectorySeparatorChar

That is not true. [The docs](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.altdirectoryseparatorchar?view=net-6.0) show that **on windows** the alt returns `/`, while the primary return `\`, but on Mac and Linux both of them return `/`. Otherwise anyone doing `path = directory + Path.DirectorySeparatorChar + filename` (say this developer didn't know about Path.Combine for some reason) would have code that does the wrong thing on Mac/Linux ([as would the BCL](https://cs.github.com/dotnet/runtime/blob/2b61381ac5ca69a5bcadbae1007afce88719d054/src/libraries/System.Private.CoreLib/src/System/IO/Path.cs#L692-L694)), which would be a pretty bad API design.